client/src/components/Clip.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import ReactPlayer from "react-player";
+
+import { useHttpClient } from "../hooks/http-hook";
+import Modal from "../UIElements/Modal";
+
+const Clip = ({
+  id,
+  title,
+  url,
+  rating,
+  removeFunc,
+  updateRating,
+  currentClip,
+  setCurrentClip,
+}) => {
+  const { isLoading, sendRequest } = useHttpClient();
+  const [modalActive, setModalActive] = useState(false);
+
+  const handleDelete = async () => {
+    try {
+      await sendRequest(`${process.env.REACT_APP_API_URL}/${id}`, "DELETE");
+      removeFunc(id);
+    } catch (error) {
+      console.error("Error deleting video:", error);
+    }
+  };
+
+  const handleHover = () => {
+    setCurrentClip({ id: id, rating: rating });
+  };
+
+  const changeRating = ({ target }) => {
+    if (target.className === "like-button") {
+      setCurrentClip({ ...currentClip, rating: currentClip.rating + 1 });
+      updateRating(currentClip.id, currentClip.rating + 1);
```

Comment:
It's interesting to need both of these calls - I would kind of expect just one function to be passed as a prop to update rating, and that would update all of the things needed - what do you think?